src/Microsoft.AspNet.Razor/TagHelpers/TagHelperDescriptor.cs

```diff
@@ -35,23 +36,64 @@ public class TagHelperDescriptor
         /// <param name="attributes">
         /// The <see cref="TagHelperAttributeDescriptor"/>s to request from the HTML tag.
         /// </param>
-        public TagHelperDescriptor([NotNull] string tagName,
-                                   [NotNull] string typeName,
-                                   [NotNull] string assemblyName,
-                                   [NotNull] IEnumerable<TagHelperAttributeDescriptor> attributes)
+        public TagHelperDescriptor(
+            [NotNull] string tagName,
+            [NotNull] string typeName,
+            [NotNull] string assemblyName,
+            [NotNull] IEnumerable<TagHelperAttributeDescriptor> attributes)
+            : this(
+                prefix: string.Empty,
+                tagName: tagName, 
+                typeName: typeName, 
+                assemblyName: assemblyName, 
+                attributes: attributes)
         {
+        }
+
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="TagHelperDescriptor"/> class with the given
+        /// <paramref name="attributes"/>.
+        /// </summary>
+        /// <param name="prefix">The prefix of the HTML element name the tag helper targets.</param>
+        /// <param name="tagName">The tag name that the tag helper targets. '*' indicates a catch-all
+        /// <see cref="TagHelperDescriptor"/> which applies to every HTML tag.</param>
+        /// <param name="typeName">The full name of the tag helper class.</param>
+        /// <param name="assemblyName">The name of the assembly containing the tag helper class.</param>
+        /// <param name="attributes">
+        /// The <see cref="TagHelperAttributeDescriptor"/>s to request from the HTML tag.
+        /// </param>
+        public TagHelperDescriptor(
+            string prefix,
+            [NotNull] string tagName,
+            [NotNull] string typeName,
+            [NotNull] string assemblyName,
+            [NotNull] IEnumerable<TagHelperAttributeDescriptor> attributes)
+        {
+            Prefix = prefix ?? string.Empty;
             TagName = tagName;
+            FullTagName = Prefix + TagName;
             TypeName = typeName;
             AssemblyName = assemblyName;
             Attributes = new List<TagHelperAttributeDescriptor>(attributes);
         }
 
         /// <summary>
+        /// The prefix of the HTML element name the tag helper targets.
+        /// </summary>
+        public string Prefix { get; private set; }
```

Comment:
FullTagName is a meta property, that's set via the ctor, it doesn't need to have a setter.